src/v/raft/consensus.cc

```diff
@@ -1653,44 +1651,39 @@ ss::future<vote_reply> consensus::do_vote(vote_request&& r) {
             // would cause subsequent votes to fail (_hbeat is updated by the
             // leader)
             _hbeat = clock_type::time_point::min();
-            return ss::make_ready_future<vote_reply>(reply);
+            co_return reply;
         }
     }
 
     // do not grant vote if log isn't ok
     if (!reply.log_ok) {
-        return ss::make_ready_future<vote_reply>(reply);
+        co_return reply;
     }
 
     if (_voted_for.id()() < 0) {
-        return write_voted_for({r.node_id, _term})
-          .then_wrapped([this, reply = std::move(reply), r = std::move(r)](
-                          ss::future<> f) mutable {
-              bool granted = false;
-
-              if (f.failed()) {
-                  vlog(
-                    _ctxlog.warn,
-                    "Unable to persist raft group state, vote not granted "
-                    "- {}",
-                    f.get_exception());
-              } else {
-                  _voted_for = r.node_id;
-                  _hbeat = clock_type::now();
-                  granted = true;
-              }
-
-              reply.granted = granted;
+        try {
+            co_await write_voted_for({r.node_id, _term});
+        } catch (...) {
+            vlog(
+              _ctxlog.warn,
+              "Unable to persist raft group state, vote not granted "
+              "- {}",
+              std::current_exception());
+            reply.granted = false;
+            co_return reply;
+        }
+        _voted_for = r.node_id;
+        reply.granted = true;
 
-              return ss::make_ready_future<vote_reply>(std::move(reply));
-          });
     } else {
         reply.granted = (r.node_id == _voted_for);
-        if (reply.granted) {
-            _hbeat = clock_type::now();
-        }
-        return ss::make_ready_future<vote_reply>(std::move(reply));
     }
+
+    if (reply.granted) {
+        _hbeat = clock_type::now();
+    }
+
+    co_return reply;
```

Comment:
looks like [this change](https://github.com/redpanda-data/redpanda/pull/11726/commits/7d68e46def71138a77cc71529039b66eb5324a3d#diff-a5f1b5c5e29537fa70312140962e3d1677e8138587d2eaa2dd0a1fcb7c396fc1R1803) is missing